counsel.el

```diff
@@ -2695,6 +2717,15 @@ This uses `counsel-ag' with `counsel-ack-base-command' replacing
 Note: don't use single quotes for the regex."
   :type 'string)
 
+(defcustom counsel-rg-version
+  (when (string-match "\\(?:\\`\\| \\)\\([^ =]+\\) " counsel-rg-base-command)
+    (let* ((rg (match-string 1 counsel-rg-base-command))
+           (version (shell-command-to-string (concat rg " --version"))))
+      (when (string-match "\\`ripgrep \\([.0-9]+\\) " version)
+        (match-string 1 version))))
+  "Ripgrep version used by `counsel-rg'."
```

Comment:
This seems like a low-level detail not fit for a `defcustom`, let alone something users should set. A more appropriate user option would be a boolean determining whether to use PCRE, don't you think?

Reply:
I moved this to the first time `counsel-rg` is called, an simplified the implementation. Let me know what you think. I would still like to handle this for the user, because otherwise it will become a point of confusion, unless users know exactly what they are doing.